src/buffer_manager.cc

```diff
@@ -78,7 +78,8 @@ void BufferManager::backup_modified_buffers()
 {
     for (auto& buf : m_buffers)
     {
-        if ((buf->flags() & Buffer::Flags::File) and buf->is_modified())
+        if ((buf->flags() & Buffer::Flags::File) and buf->is_modified()
+            and !(buf->flags() & Buffer::Flags::ReadOnly))
```

Comment:
can you replace the `!` there with `not`, in Kakoune source code we use the alternative, word based logical operators.